iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsGenerationCheck.java

```diff
@@ -87,17 +91,42 @@ public class SecretsGenerationCheck implements IacCheck {
 
   private static final CommandDetector SSHPASS_P_FLAG_SPACE_PWD = commandFlagSpace("sshpass", "-p");
 
-  private static final CommandDetector SSHPASS_P_FLAG_NO_SPACE_PWD = CommandDetector.builder()
-    .with("sshpass")
-    .withAnyIncludingUnresolvedExcluding(arg -> !arg.startsWith("-p"))
-    .withArgumentResolutionIncludeUnresolved(resolution -> resolution.value().startsWith("-p") &&
-      (resolution.value().length() > 2 || resolution.argument().expressions().size() > 1))
-    .build();
+  private static final CommandDetector SSHPASS_P_FLAG_NO_SPACE_PWD = commandsFlagNoSpacePwd(List.of("sshpass"), "-p");
+
+  private static final List<String> MYSQL_COMMANDS = List.of("mysql", "mysqladmin", "mysqldump");
+  private static final CommandDetector MYSQL_PASSWORD_EQUALS_PWD = commandsFlagEquals(MYSQL_COMMANDS, PASSWORD_FLAG);
+
+  private static final CommandDetector MYSQL_P_FLAG_NO_SPACE_PWD = commandsFlagNoSpacePwd(MYSQL_COMMANDS, "-p");
+
+  private static CommandDetector commandsFlagNoSpacePwd(List<String> commands, String flag) {
```

Comment:
Maybe rename `commandsFlagNoSpacePwd` to `commandsFlagNoSpace`, as this method does not seem to be dedicated to password flags only.